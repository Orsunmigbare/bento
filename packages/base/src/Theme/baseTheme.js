import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
