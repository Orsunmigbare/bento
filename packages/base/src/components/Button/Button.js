/**
 * @class Button
 */
import PropTypes from "prop-types"
import React from "react"
import styled from "styled-components"
import {
  targetSize,
  spacing,
  borderRadius,
  font,
  color,
  themify
} from "@staccx/theme"
const tinycolor = require("tinycolor2")

export const BUTTON = "Button"
export const ButtonStyle = css`
  background-color: ${color("primary")};
  color: ${p =>
    tinycolor
      .mostReadable(color("primary")(p), ["#fff"], {
        includeFallbackColors: true,
        level: "AA",
        size: "large"
      })
      .toString()};
  border-width: 0;
  border-radius: ${borderRadius()};
  padding: ${spacing("tiny")} ${spacing("medium")};
  cursor: pointer;
  min-height: ${targetSize()};
  text-decoration: none;
  text-align: center;
  display: inline-block;
  font-size: ${font("base")};
  font-family: ${font("body", "type")};
  line-height: 32px;
  font-weight: 600;
  white-space: nowrap;
  justify-content: center;
  align-items: center;
  margin-right: ${spacing("small")};
  margin-bottom: ${spacing("small")};
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  &:hover,
  &:focus {
    outline: none;
    background-color: ${p =>
      tinycolor(color("primary")(p))
        .darken(5)
        .toString()};
    color: ${p =>
      tinycolor
        .mostReadable(tinycolor(color("primary")(p)).darken(5), ["#fff"], {
          includeFallbackColors: true,
          level: "AA",
          size: "large"
        })
        .toString()};
  }

  &:last-child,
  &:last-of-type {
    margin-right: 0;
  }

  &:last-child:first-child {
    margin-bottom: 0;
  }

  ${themify(BUTTON)};
`
const Button = styled.button`
  ${ButtonStyle};
`

const defaultProps = {
  onClick: null,
  to: "#",
  className: ""
}

const propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element,
    PropTypes.string,
    PropTypes.array
  ]).isRequired,
  onClick: PropTypes.func,
  className: PropTypes.string,
  to: PropTypes.string
}

Button.defaultProps = defaultProps
Button.propTypes = propTypes

/** @component */
export default Button
