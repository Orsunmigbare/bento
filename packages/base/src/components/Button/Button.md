Button

```jsx harmony
<Button>Tester</Button>
```
