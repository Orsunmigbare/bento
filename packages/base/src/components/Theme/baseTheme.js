export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
