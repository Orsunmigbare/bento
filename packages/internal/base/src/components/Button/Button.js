import PropTypes from "prop-types"
import React from "react"
import styled from "styled-components"
import {
  targetSize,
  spacing,
  borderRadius,
  fontFamily,
  fontWeight,
  font,
  color,
  themify
} from "@staccx/theme"
import themePropTypes from "../constants/themePropTypes"
const tinycolor = require("tinycolor2")

const tProps = {
  button: {
    name: "Button",
    description: "Button styles",
    type: themePropTypes.style
  }
}

const ButtonComp = styled.button`
  background-color: ${color.primary};
  color: ${p =>
    tinycolor
      .mostReadable(color.primary()(p), ["#fff"], {
        includeFallbackColors: true,
        level: "AA",
        size: "large"
      })
      .toString()};
  border-width: 0;
  border-radius: ${borderRadius};
  padding: ${spacing.tiny} ${spacing.medium};
  cursor: pointer;
  min-height: ${targetSize.normal};
  text-decoration: none;
  text-align: center;
  display: inline-block;
  font-size: ${font.base};
  font-family: ${fontFamily.body};
  line-height: 32px;
  font-weight: ${fontWeight.bold};
  white-space: nowrap;
  justify-content: center;
  align-items: center;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  transition: background 0.2s ease;

  &:hover,
  &:focus {
    outline: none;
    background-color: ${p =>
      tinycolor(color.primary()(p))
        .darken(5)
        .toString()};
    color: ${p =>
      tinycolor
        .mostReadable(tinycolor(color.primary()(p)).darken(5), ["#fff"], {
          includeFallbackColors: true,
          level: "AA",
          size: "large"
        })
        .toString()};
  }

  ${themify(tProps.button.name)};
`
/**
 * Button component. Use to click stuff
 */
const Button = props => <ButtonComp {...props} />

const defaultProps = {
  onClick: null,
  to: "#",
  className: ""
}

const propTypes = {
  /**
   * The children of the lamb
   */
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element,
    PropTypes.string,
    PropTypes.number,
    PropTypes.array
  ]).isRequired,
  onClick: PropTypes.func,
  className: PropTypes.string,
  to: PropTypes.string,
  test: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
}

Button.defaultProps = defaultProps
Button.propTypes = propTypes
Button.themeProps = tProps

export default Button
