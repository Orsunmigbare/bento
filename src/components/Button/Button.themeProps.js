import { themePropTypes } from "../../theming"

const themeProps = {
  button: {
    name: "Button",
    description: "Button styles",
    type: themePropTypes.style
  }
}
export default themeProps
