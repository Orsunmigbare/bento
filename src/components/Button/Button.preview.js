import React from "react"
import Button from "./Button"

const preview = {
  title: "Button",
  category: "Components/Buttons",
  tags: ["component", "clickable", "button"],
  component: Button,
  render: props => <Button {...props} />
}

export default preview
