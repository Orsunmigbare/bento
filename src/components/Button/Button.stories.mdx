import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Button from "./Button"

<Meta title="Components|Button" component={Button}/>


# Button
Forklaring p√• hva det er og hvordan den kan brukes


# import

```javascript
Import {Button} from "@staccx/bento"
```

# Types of buttons


## Standard Button
This is the standard button



<Preview>
  <Story name="Standard">
   <Button>Knapp</Button>
  </Story>
</Preview>


## Two buttons
check console for onClick event

<Preview>
  <Story name="Two Buttons">
  <>
   <Button>Knapp</Button>
   <Button>Knapp</Button>
   </>
  </Story>
</Preview>

## Long String
<Preview>
  <Story name="Long String">
   <Button>KnappKnappKnappKnappKnappKnappKnappKnappKnappKnappKnappKnappKnappKnappKnapp</Button>
  </Story>
</Preview>



# Props for button
<Props of={Button} />
