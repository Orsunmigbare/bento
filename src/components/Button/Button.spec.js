import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../theming/themes/baseTheme"
import Button from "./Button"

const text = "Test"

describe("Button", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Button>{text}</Button>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("with onClick", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Button onClick={() => console.log("Clicked")}>{text}</Button>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("no children", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Button onClick={() => console.log("Clicked")} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
