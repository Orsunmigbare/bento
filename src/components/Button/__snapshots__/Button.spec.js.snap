// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshots No props 1`] = `
<button
  className="sc-AxjAm ecAfij"
  onClick={null}
>
  Test
</button>
`;

exports[`Button Snapshots no children 1`] = `
<button
  className="sc-AxjAm ecAfij"
  onClick={[Function]}
>
  button
</button>
`;

exports[`Button Snapshots with onClick 1`] = `
<button
  className="sc-AxjAm ecAfij"
  onClick={[Function]}
>
  Test
</button>
`;
